src/app/ReadClient.h

```diff
@@ -351,11 +351,17 @@ class ReadClient : public Messaging::ExchangeDelegate
      *  Re-activate an inactive subscription.
      *
      *  When subscribing to LIT-ICD and liveness timeout reached and OnResubscriptionNeeded returns
-     * CHIP_ERROR_LIT_SUBSCRIBE_INACTIVE_TIMEOUT, the read client will move to the InactiveICDSubscription state and
-     * resubscription can be triggered via OnActiveModeNotification().
+     *  CHIP_ERROR_LIT_SUBSCRIBE_INACTIVE_TIMEOUT, the read client will move to the InactiveICDSubscription state and
+     *  resubscription can be triggered via OnActiveModeNotification().
      *
      *  If the subscription is not in the `InactiveICDSubscription` state, this function will do nothing. So it is always safe to
-     * call this function when a check-in message is received.
+     *  call this function when a check-in message is received.
+     *
+     *  If the server sends out check-in message, and there is a active tracked active subscription in client side at the same time,
```

Comment:
There's a lot of verbiage here, but it's all very confusing and mostly describes implementation details not API.

The API is: This function should be called when the peer is an ICD that is checking in and this ReadClient represents a subscription that would cause that ICD to not need to check in anymore.

no?  That's about all the documentation this function needs.

Reply:
updated, thanks